en/diagnostics/index.md

```diff
@@ -1,9 +1,29 @@
 # Stride diagnostics
 
-Some C# compiler errors have corresponding topics that explain why the error is generated, and, in some cases, how to fix the error. Use one of the following steps to see whether help is available for a particular error message.
+Stride.Core.CompilerServices contains Roslyn code analyzers.
 
-[Rule of Thumb Serialization](https://github.com/stride3d/stride-docs/blob/master/en/manual/scripts/serialization.md#rule-of-thumb)
+> [!IMPORTANT]
+> These analyzers do not perform any kind of telemetry. The same analyzers get used by your IDE when you get a CSXXXX Diagnostic
+
+They analyze the code for possible issues in your project with the Stride.Core design.
+To avoid unexpected runtime/compile time/editor time behaviour these analyzers try to warn as soon as possible for issues that may occur.
+Each of the following pages contain information about Diagnostic codes that can be reported by the Roslyn analyzers, which are built into Stride.Core.CompilerServices.
+
+The Information covers:
+    - When is the Diagnostics reported
+    - An explanation why it is necessary to report the Diagnostic
+    - Examples when such a Diagnostics occurs
+    - Information about how to resolve the Diagnostic
+
+If an error is reported it is possible to click in the IDE on the Diagnostic code.
```

Comment:
Here maybe make it bold **Diagnostic code** if you are referring to the menu in the IDE, also the capital D is ok, because it is a menu item.